### PR TITLE
AuthSlice ThunkAPI calls

### PR DESCRIPTION
**Pull request by t007rushi**

### CODE REVIEW

File: src/features/auth/authSlice.js [hunk 1]
```diff
@@ -1,35 +1,286 @@
...
+  signOutStatus: "idle",
+  getUserStatus: "idle",
 };
 
+//signin
+export const SignIn = createAsyncThunk(
+  "auth/SignIn",
+  async ({ email, password }) => {
+    try {
+      console.log(email, password);
```
Comment (sushil-mod) [inline src/features/auth/authSlice.js]: remove console

File: src/features/auth/authSlice.js [hunk 2]
```diff
@@ -1,35 +1,286 @@
...
+    console.log(error.message);
+  }
+});
+
+//twitter auth
+export const handleTLogin = createAsyncThunk("auth/TwitterSignIn", async () => {
+  try {
+    const Tprovider = new TwitterAuthProvider();
+    await signInWithPopup(auth, Tprovider).then((userCred) => {
+      console.log(userCred)});
```
Comment (sushil-mod) [inline src/features/auth/authSlice.js]: if no future use plz remove console

File: src/features/auth/authSlice.js [hunk 3]
```diff
@@ -1,35 +1,286 @@
...
+    console.log(error.message);
+  }
+});
+
+//facebook auth
+export const handleFLogin = createAsyncThunk("auth/FacebookSignIn", async () => {
+  try {
+    const Fprovider = new FacebookAuthProvider();
+    await signInWithPopup(auth, Fprovider).then((userCred) => {
+      console.log(userCred)
```
Comment (sushil-mod) [inline src/features/auth/authSlice.js]: here too

File: src/features/auth/authSlice.js [hunk 4]
```diff
@@ -1,35 +1,286 @@
...
+  async ({ firstName, lastName, email, password }) => {
+    try {
+      const auth = getAuth(app);
+      const data = await createUserWithEmailAndPassword(auth, email, password);
+
+      await setDoc(doc(database, "users", data.user.uid), {
+        firstName,
+        lastName,
+        email,
+        id: data.user.uid,
```
Comment (JantuDeb) [inline src/features/auth/authSlice.js:105]: create a user object then you don't need to duplicate this in return

**Review by sushil-mod — APPROVED** [review]
some console log has to remove rest code looks good. Also, like your onclickoutside hooks

**Review by JantuDeb — APPROVED** [review]
- Good coding conventions.
- clean and structure code
-

File: src/components/CreatePost.jsx [hunk 5]
```diff
@@ -1,33 +1,47 @@
...
   const handleSubmit = () => {
     addDoc(collectionRef, {
-      content: postData.content,
+      description: postData.description,
+      title: postData.title,
+      user: user.firstName,
     })
-      .then(() => alert("Data Added"))
+      .then(() => {})
       .catch((err) => console.log(err));
```
Comment (narharikale) [inline src/components/CreatePost.jsx:21]: console.error should be more appropriate

**Review by narharikale — APPROVED** [review]
great work rushi.
good to go 🤩

**Pull request merged**